src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -125,65 +133,142 @@ const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelec
+      }
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_LEFT) {
+      event.preventDefault();
+      if (document.getElementById(currentNodeId).ariaExpanded === 'true') {
+        handleOnClick(event, item);
+      } else if (!document.getElementById(currentNodeId).ariaHasPopup || !document.getElementById(currentNodeId).ariaExpanded || document.getElementById(currentNodeId).ariaExpanded === 'false') {
+        findParent();
+      }
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_HOME) {
+      event.preventDefault();
+      cursor.current = 0;
+      setCurrentNodeId(visibleNodes.current[cursor.current].id);
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_END) {
+      event.preventDefault();
+      cursor.current = visibleNodes.current.length - 1;
+      setCurrentNodeId(visibleNodes.current[cursor.current].id);
+    } else if (event.nativeEvent.keyCode >= KeyCode.KEY_A && event.nativeEvent.keyCode <= KeyCode.KEY_Z) {
+      event.preventDefault();
+      const char = String.fromCharCode(event.nativeEvent.keyCode);
+      findNode(char);
+    } else if (event.nativeEvent.shiftKey && event.nativeEvent.keyCode === KeyCode.KEY_TAB) {
+      handleUpArrow();
+    } else if (event.nativeEvent.keyCode === KeyCode.KEY_TAB) {
+      handleDownArrow();
     }
   };
 
-  const renderMenuItems = (currentMenuItem, firstLevel) => {
+  const renderMenuItems = (currentMenuItem, parent = '', firstLevel = false) => {
     if (!currentMenuItem) {
       return undefined;
     }
 
     return currentMenuItem.map((item) => {
       const itemIsOpen = openKeys[item.path];
       const itemHasChildren = item.childItems !== undefined;
+      const id = item.name.split(' ').join('-');
       let isSelected = false;
       let selectedRef;
 
       if (selectedPath === item.path) {
         isSelected = true;
         selectedRef = selectedItem;
+      } else if (currentNodeId === id) {
+        isSelected = true;
       }
 
+      visibleNodes.current.push({ id, parent });
       const menuItemClassNames = classNames(
         cx([
           'item',
           { 'is-selected': isSelected },
         ]),
       );
 
+      const optionalAttributes = itemHasChildren ? { 'aria-expanded': itemIsOpen, 'aria-haspopup': true } : {};
+
       return (
         <React.Fragment key={item.path}>
           <div className={!firstLevel ? cx('indent') : null}>
             <div
               className={menuItemClassNames}
               tabIndex="0"
-              role="link"
-              id={item.name.split(' ').join('-')}
-              aria-haspopup={itemHasChildren}
+              role="treeitem"
+              id={id}
               onKeyDown={event => handleKeyDown(event, item)}
               onClick={event => handleOnClick(event, item)}
               onBlur={enableFocusStyles}
               onMouseDown={disableFocusStyles}
               data-focus-styles-enabled
               ref={selectedRef}
+              {...optionalAttributes}
```

Comment:
Can we move the optionalAttributes into the props directly?

```suggestion
              'aria-expanded': itemHasChildren  && itemIsOpen
              aria-haspopup': itemIsOpen
```

Reply:
My understanding of `aria-expanded` is that it is used to indicate a node that has child nodes, `false` shows that it is not currently expanded `true` indicating that it is expanded. I thought that having `aria-expanded: false` on end nodes would be confusing to a user using a screen reader, since they do not have child nodes and cannot be expanded. Doing it this way prevents attaching the `aria-expanded` attribute entirely to nodes with no children.